### DISCUSSION TITLE
Improve pedestrian costing

### DISCUSSION THREAD

**Discussion by nilsnolde** [valhalla/valhalla#3838] [Ideas]
Let's collect some ideas what can be improved with pedestrian routing. We can see which ones actually make sense to support and create single tickets for those. I don't really have that many, but I'm sure there's lots of ideas, Cc @georg-d 

## Add
- use `lit` OSM tag (used on 8 Mio ways!). That's actually an easy one and could be done by the community.
- avoiding higher-class roads both as safety and "pleasure" measure, maybe with a similar logic as in bicycle costing, also using `use_roads`?
- a huge one, but I'll still leave it here: a "green" costing option favoring walking/biking along/through parks/green areas. Actually, at least in Berlin, Google Maps semi-recently has added that option per default and it's really nice! But we can do it much nicer than Google of course, can't opt out of preferring green at all. One of my favorite things I'd like to work on. It's quite a big one though.

Maybe it'd make sense to e.g. add a costing option "safe" which includes things like `lit` and penalize higher-class roads, esp those without a sidewalk (we already favor sidewalk roads, but don't penalize road class).

## Change
- `destination_only` transition penalty? Is that relevant for peds? Not sure what's all conflated into Valhalla's representation of dest_only, probably more than just "access=destination" like tags? The classic use of "Anlieger frei", at least in Germany, doesn't apply to peds.
- `living_street` penalty? Sounds like that should be implicitly encouraged to take rather than discouraged by costing option. I mean it's fine to respect the costing option, but IMO it should still be preferred by default, not ignored by default.

**Comment by georg-d** [comment]
Allowing to **prefer / discourage surface types and maybe smoothness** could be of interest – for jogging, a softer surface is usually preferred, on wet days "cleaner" surfaces like concrete, asphalt etc are usually preferred, some people (skaters, health limitations, etc) might be interested in as smooth as possible. I'm naming here both the router as well as the UI part, so it's easier to get the idea. Because of the sheer amount of surface types, a setting for each single type will be quite an effort for users and thus probably not be used often, but groupings like "soft", "not muddy", "hard & smooth" etc seem convenient. Optimal solution coming to my mind now is like with many software installers, i.e. 

- a choice to choose such a group (drop down? list of radio buttons?) 
- a possibility to unfold/popup the detailed list with checkbox or numeric factor for each single surface type, so users could fine tune according to their region and activity/need (e.g. besides the drop down a triangle or gear icon acting as a button)
- a possibility to save the detailed settings (e.g. as download / copy to clipboard, but also add a named group and store it in a cookie)

**Comment by dnesbitt61** [comment]
Surface tags in OSM are consolidated into 7 different surface types (and an 8th for surfaces that may make a way unpassable):
```
enum class Surface : uint8_t {
  kPavedSmooth = 0,
  kPaved = 1,
  kPavedRough = 2,
  kCompacted = 3,
  kDirt = 4,
  kGravel = 5,
  kPath = 6,
  kImpassable = 7
};
```
Being able to select specific surface tags (or groups of them) for favoring/avoiding is not possible with the current data specification.
